### PR TITLE
New release and publish process

### PR DESCRIPTION
**Pull request by florian-vuillemot**

### CODE REVIEW

**Comment by dinhlongviolin1** [comment]
With the new release process, the version would have to be provided in the input of workflow_dispatch right

**Comment by florian-vuillemot** [comment]
Yes. The version should be the same than in the setup.py + major & minor should match the branch name ("release/Major.Minor")

**Comment by florian-vuillemot** [comment]
@FabienLelaquais @dinhlongviolin1 important point. The release does not build the web app anymore. So, if a user installs taipy from a Github release, npm has to be installed.
The behaviour of Pypi is not the same because Pypi wants a package already built. And the web app generation is done by the build.

**Review by FabienLelaquais — APPROVED** [review]
Looks good to me

**Comment by FabienLelaquais** [comment]
> @florian-vuillemot I'm not sure I understand the npm build part of this though.
A release must contain the WebApp, and users should never have to have npm installed locally.
Are we on the same page?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,81 +1,79 @@
...
+          rm -rf taipy
+          pip install dist/*.tar.gz
+          pip install pytest pytest-mock
+          pytest
 
-    runs-on: ${{ matrix.os }}
+  release:
+    needs: [test-package]
+    timeout-minutes: 20
+    name: Release taipy-core Package
```
Comment (dinhlongviolin1) [inline .github/workflows/release.yml]: taipy-gui

**Review by dinhlongviolin1 — APPROVED** [review]
Looks solid to me

File: .github/workflows/release.yml (hunk 1 above)
Comment (florian-vuillemot) [inline .github/workflows/release.yml]:
```suggestion
    name: Release package
```

**Pull request merged**